tools/sdk-generation-pipeline/packages/sdk-generation-lib/src/lib/publishResult.ts

```diff
@@ -0,0 +1,241 @@
+        await taskResultDao.put(pipelineBuildId, taskResult);
+    }
+
+    const codeGenerationDao: CodeGenerationDao = new CodeGenerationDaoImpl(
+        mongoDbConnection
+    );
+    const cg: CodeGeneration = new CodeGeneration();
+    cg.name = sdkGenerationName;
+    cg.service = service;
+    cg.serviceType = serviceType;
+    cg.tag = tag !== undefined ? tag : '';
+    cg.sdk = language;
+    cg.swaggerRepo = swaggerRepo;
+    cg.sdkRepo = sdkRepo;
+    cg.codegenRepo = codegenRepo;
+    cg.owner = owner === undefined ? '' : owner;
+    cg.type = triggerType;
+    cg.status = status;
+    cg.lastPipelineBuildID = pipelineBuildId;
+    cg.swaggerPR = swaggerPR === undefined ? '' : swaggerPR;
+    cg.codePR = codePR === undefined ? '' : codePR;
+    await codeGenerationDao.submitCodeGeneration(cg);
+
+    await mongoDbConnection.close();
+}
+
+export async function uploadLogsAndResult(
+    logsAndResultPath: string,
+    pipelineBuildId: string,
+    taskName: AzureSDKTaskName,
+    sdkGenerationName: string,
+    azureStorageBlobSasUrl: string,
+    azureBlobContainerName: string
+) {
+    let stepName: string = taskName;
+    if (logsAndResultPath === undefined) {
+        throw new Error(`logsAndResultPath is empty`);
+    }
+    if (pipelineBuildId === undefined) {
+        throw new Error(`pipelineBuildId is empty`);
+    }
+    if (sdkGenerationName === undefined) {
+        throw new Error(`sdkGenerationName is empty`);
+    }
+    if (azureStorageBlobSasUrl === undefined) {
+        throw new Error(`azureStorageBlobSasUrl is empty`);
+    }
+    if (azureBlobContainerName === undefined) {
+        throw new Error(`azureBlobContainerName is empty`);
+    }
+    if (taskName === undefined) {
+        logger.info(`taskName is undefined, set full log`);
+        stepName = 'full';
+    }
+
+    const azureBlobClient = new AzureBlobClient(
+        azureStorageBlobSasUrl,
+        azureBlobContainerName
+    );
+    const logsAndResultPathArray = JSON.parse(logsAndResultPath);
+    for (const file of logsAndResultPathArray) {
+        if (fs.existsSync(file)) {
+            let blobName: string = file.includes('.json')
+                ? `${pipelineBuildId}/${sdkGenerationName}-${stepName}-result.json`
+                : `${pipelineBuildId}/${sdkGenerationName}-${stepName}.log`;
+            await azureBlobClient.publishBlob(file, blobName);
+            logger.info(`Publish ${file} Success !!!`);
+        } else {
+            logger.error(`Publish result failed !, ${file} is missed`);
+        }
+    }
+}
+
+function getFileListInPackageFolder(packageFolder: string) {
+    child_process.execSync('git add .', {
+        encoding: 'utf8',
+        cwd: packageFolder,
+    });
+    const files = child_process
+        .execSync('git ls-files', { encoding: 'utf8', cwd: packageFolder })
+        .trim()
+        .split('\n');
+
+    return files;
+}
+
+export async function uploadGenerateAndBuildFile(
```

Comment:
Do not in one function to send sourceCode and Artifacts the same time. Should separate it.

Reply:
done